api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -538,6 +543,17 @@ func (m *AzureManagedControlPlane) validateVirtualNetworkUpdate(old *AzureManage
 	return allErrs
 }
 
+// validateNetworkPluginModeUpdate validates update to NetworkPluginMode.
+func (m *AzureManagedControlPlane) validateNetworkPluginModeUpdate(old *AzureManagedControlPlane) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if ptr.Deref(m.Spec.NetworkPluginMode, "") == NetworkPluginModeOverlay && old.Spec.NetworkPolicy != nil {
```

Comment:
I verified that AKS does allow this on create. It only errors on updates.

Reply:
Wut

We're saying:

1. You can create a cluster w/ CNI Overlay + NetworkPolicy when you create your cluster
2. You cannot update a cluster w/ NetworkPolicy configured from non-CNI Overlay to CNI Overlay
3. You can update a cluster to CNI Overlay if NetworkPolicy is not enabled

?

Reply:
Correct

Reply:
I wonder if scenario #1 actually works? Probably out-of-scope for this PR to definitely prove that, so we can take the API at its word and implement CAPZ accordingly.

Thanks for doing the extra validation here.